### ISSUE TITLE
Thank you for opening your code

### ISSUE THREAD

**Issue by ElleZWQ** [GasolSun36/MVP#1]
## Thank you for opening your code, your work is very interesting. I have two questions:
**First question:**

In MVP_llava.py:
I change this part:

if answer_lower in ["yes", "no"]:
     diff_prob = top_two_probs.values[0] - top_two_probs.values[1]
     if answer_lower == "yes":
          aggregation_dict['Yes'] += value * diff_prob
    elif answer_lower == "no":
           aggregation_dict['No'] += value * diff_prob

To following part:

if answer_lower == "yes":
           aggregation_dict['Yes'] += value * diff_prob
elif answer_lower == "no":
          aggregation_dict['No'] += value * diff_prob

It was no_aggretion_logits and yes_aggretion_logits, but they were not used. Based on the description of your paper, I have changed it to the above form.

**Second question:**

I change this part of eval_pope.py

def compute_probabilities(weights, *args):
    return sum(weight * data['aggretion_logits'][key] for weight, data, key in zip(weights, args, ['Yes', 'No']))

To following part:

def compute_probabilities(weights, *args):
    sum = {key: 0 for key in ['Yes','No']}
    for weight,data,key in zip(weights,args,['Yes','No']):
        print(weight)
        print(data['aggretion_logits'][key])
        sum[key] += weight * data['aggretion_logits'][key]
       return sum['Yes'],sum['No']

because  yes_prob, no_prob = compute_probabilities(weights, bottom_up, top_down, normal) have two outputs.

**After changing the above parts, I found that the accuracy was 41.43%, which is very different from your presentation.
Could you please help me check if I have corrected your code reasonably?**

Thank you!

**Comment by GasolSun36** [comment]
> ## Thank you for opening your code, your work is very interesting. I have two questions:
> **First question:**
> 
> In MVP_llava.py: I change this part:
> 
> if answer_lower in ["yes", "no"]: diff_prob = top_two_probs.values[0] - top_two_probs.values[1] if answer_lower == "yes": aggregation_dict['Yes'] += value * diff_prob elif answer_lower == "no": aggregation_dict['No'] += value * diff_prob
> 
> To following part:
> 
> if answer_lower == "yes": aggregation_dict['Yes'] += value * diff_prob elif answer_lower == "no": aggregation_dict['No'] += value * diff_prob
> 
> It was no_aggretion_logits and yes_aggretion_logits, but they were not used. Based on the description of your paper, I have changed it to the above form.
> 
> **Second question:**
> 
> I change this part of eval_pope.py
> 
> def compute_probabilities(weights, *args): return sum(weight * data['aggretion_logits'][key] for weight, data, key in zip(weights, args, ['Yes', 'No']))
> 
> To following part:
> 
> def compute_probabilities(weights, *args): sum = {key: 0 for key in ['Yes','No']} for weight,data,key in zip(weights,args,['Yes','No']): print(weight) print(data['aggretion_logits'][key]) sum[key] += weight * data['aggretion_logits'][key] return sum['Yes'],sum['No']
> 
> because yes_prob, no_prob = compute_probabilities(weights, bottom_up, top_down, normal) have two outputs.
> 
> **After changing the above parts, I found that the accuracy was 41.43%, which is very different from your presentation. Could you please help me check if I have corrected your code reasonably?**
> 
> Thank you!



> ## Thank you for opening your code, your work is very interesting. I have two questions:
> **First question:**
> 
> In MVP_llava.py: I change this part:
> 
> if answer_lower in ["yes", "no"]: diff_prob = top_two_probs.values[0] - top_two_probs.values[1] if answer_lower == "yes": aggregation_dict['Yes'] += value * diff_prob elif answer_lower == "no": aggregation_dict['No'] += value * diff_prob
> 
> To following part:
> 
> if answer_lower == "yes": aggregation_dict['Yes'] += value * diff_prob elif answer_lower == "no": aggregation_dict['No'] += value * diff_prob
> 
> It was no_aggretion_logits and yes_aggretion_logits, but they were not used. Based on the description of your paper, I have changed it to the above form.
> 
> **Second question:**
> 
> I change this part of eval_pope.py
> 
> def compute_probabilities(weights, *args): return sum(weight * data['aggretion_logits'][key] for weight, data, key in zip(weights, args, ['Yes', 'No']))
> 
> To following part:
> 
> def compute_probabilities(weights, *args): sum = {key: 0 for key in ['Yes','No']} for weight,data,key in zip(weights,args,['Yes','No']): print(weight) print(data['aggretion_logits'][key]) sum[key] += weight * data['aggretion_logits'][key] return sum['Yes'],sum['No']
> 
> because yes_prob, no_prob = compute_probabilities(weights, bottom_up, top_down, normal) have two outputs.
> 
> **After changing the above parts, I found that the accuracy was 41.43%, which is very different from your presentation. Could you please help me check if I have corrected your code reasonably?**
> 
> Thank you!

Thank you for your interest in our work!

1. In Formula 4 of our paper, the expression 'diff_prob = top_two_probs.values[0] - top_two_probs.values[1]' quantifies the confidence of the LLM in the current decoded token.
2. As clearly stated in our paper, our final voting mechanism involves a weighted average of the 'yes' and 'no' tokens from each perspective. Therefore, I believe the code you provided is unnecessary.

**Comment by ElleZWQ** [comment]
Thank you for your reply.
I believe you misunderstood my questions. 
I want to explain that I didn't change your idea.
I just want to say the current version of your code can not run **because of some grammar mistakes.** You can run it by yourself.
The above changes are proposed to solve these mistakes.
Additionally, after I corrected the following evaluation part of eval_pope.py, there were also grammar mistakes. I got similar scores as your paper showed.
<img width="686" alt="image" src="https://github.com/user-attachments/assets/b06e8334-c719-4d3e-b641-fc202900c858">

The code in this image is yours. 
Thank you.

**Comment by GasolSun36** [comment]
Hi

> Thank you for your reply. I believe you misunderstood my questions. I want to explain that I didn't change your idea. I just want to say the current version of your code can not run **because of some grammar mistakes.** You can run it by yourself. The above changes are proposed to solve these mistakes. Additionally, after I corrected the following evaluation part of eval_pope.py, there were also grammar mistakes. I got similar scores as your paper showed. <img alt="image" width="686" src="https://private-user-images.githubusercontent.com/149352957/367310308-b06e8334-c719-4d3e-b641-fc202900c858.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYyMzQwODUsIm5iZiI6MTcyNjIzMzc4NSwicGF0aCI6Ii8xNDkzNTI5NTcvMzY3MzEwMzA4LWIwNmU4MzM0LWM3MTktNGQzZS1iNjQxLWZjMjAyOTAwYzg1OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxM1QxMzIzMDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01Y2M1MTEzYmI5MzhmYzNhNmFlYzUyYjkyM2QwYmRiMzMyZTkyM2MzNmM2YjFjYmM0YWM0ZDQ0NjQwNDJkZmRmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.uu0jn_qst2Cao0CeYH4a_PGRrGAJwIN-6Gaske4NUYA">
> 
> The code in this image is yours. Thank you.

Hi, 

I'm terribly sorry that I misunderstand your question. I will check the code carefully (because I have been too busy at work recently, this code was sorted out a long time ago, and there are some implementation details I need to check as well.)

Anyway, thanks a lot for your help!

**Comment by ElleZWQ** [comment]
Ok! I know you are busy! It doesn't matter. I appreciate your reply.
